spec-0007/index.md

```diff
@@ -5,69 +5,69 @@ author:
+We recommend:
+
+- avoiding the use of global state and legacy bitstream generators, and
+- standardizing the usage and interpretation of an `rng` keyword for seeding.
+
+We suggest implementing these principles by:
 
-- Standardize the usage and interpretation of an `rng` keyword for seeding, and
-- avoid the use of global state and legacy bitstream generators.
+- deprecating the use of `numpy.random.seed` to control the random state,
+- deprecating the use of `random_state`/`seed` arguments in favor of an `rng` argument in all functions that require random number generation, and
+- using `numpy.random.default_rng` to validate the `rng` argument and instantiate a `Generator`.
 
-We suggest doing this by:
+### Scope
 
-- Passing the `rng` argument to `numpy.random.default_rng` to instantiate a `Generator`, and
-- deprecating the use of `np.seed` and `RandomState`.
+This is intended as a recommendation to all libraries that allow users to control the
+state of a NumPy random number generator. It is specifically targeted toward functions
+that allow use of `numpy.random.seed` to control the random state and accept
+`RandomState` instances via an argument with a name other than `rng`, but the ideas
+are more broadly applicable. Use of random number generators other than those provided
+by NumPy are not considered at this time, but may be addressed in a future version.
 
 ### Concepts
 
-- `BitGenerator`: Generates a stream of pseudo-random bits. The default generator in NumPy (`np.random.default_rng`) uses PCG64.
+- `BitGenerator`: Generates a stream of pseudo-random bits. The default generator in NumPy (`numpy.random.default_rng`) uses PCG64.
 - `Generator`: Derives pseudo-random numbers from the bits produced by a `BitGenerator`.
 - `RandomState`: a [legacy object in NumPy](https://numpy.org/doc/stable/reference/random/index.html), similar to `Generator`, that produces random numbers based on the Mersenne Twister.
 
 ### Constraints
 
 NumPy, SciPy, scikit-learn, scikit-image, and NetworkX all implement pseudo-random seeding in slightly different ways.
 Common keyword arguments include `random_state` and `seed`.
-In practice, the seed is unfortunately also often controlled using `np.random.seed`.
+In practice, the seed is unfortunately also often controlled using `numpy.random.seed`.
 
 ## Implementation
 
-<!--
-Discuss how this would be implemented.
--->
-
 Legacy behavior in packages such as scikit-learn (`sklearn.utils.check_random_state`) typically handle `None` (use the global seed state), an int (convert to `RandomState`), or `RandomState` object.
 
 Two strong motivations for moving over to `Generator`s are:
 
 (1) they avoid naïve seeding strategies, such as using successive integers, via the underlying [SeedSequence](https://numpy.org/doc/stable/reference/random/parallel.html#seedsequence-spawning);
-(2) they avoid using global state (from `np.random.mtrand._rand`).
+(2) they avoid using global state (from `numpy.random.mtrand._rand`).
```

Comment:
Just making sure - you intended for all of this to show up as a paragraph without line breaks, right?

Reply:
I will change that.

Reply:
Done (locally).